### DISCUSSION TITLE
Installing programs in Cosmos

### DISCUSSION THREAD

**Discussion by smlaticc** [CosmosOS/Cosmos#2679] [Q&A]
Hi there.
Currently new in Cosmos. I'm curious about installing apps in os created with cosmos.
Is there a way to install for example Google Chrome in os?

**Comment by terminal-cs** [comment]
No, there isn't, you need to make a program installer and manager. Additionally you need to create an elf loader and linker, which also requires paginh

**Reply by smlaticc** [reply]
Is out there some examples for that?
Can You share some urls?

**Reply by terminal-cs** [reply]
> Is out there some examples for that?
> Can You share some urls?

This will take years of programming and learning, I dont even know it myself. Additionally you will need to implement all of the windows or Linux syscalls to get it working

**Reply by terminal-cs** [reply]
You can find sources on the osdev wiki

**Comment by winbamstudios** [comment]
the fact that someone who's trying to make their own OS LEGIT thinks they can port chrome. i want to cry.

**Reply by smlaticc** [reply]
Read my question again, and if it still makes you cry, than you should stop programming cause your logic is in mess

**Reply by winbamstudios** [reply]
Oh.

**Comment by tacitmarmot271** [comment]
ive got a program manager that i could post for you

**Comment by Yarpopcat08** [comment]
Nah, only if You implement all the stuff needed for executables to run. No existing apps could be installed (if You make Your own format), so no Chrome for You, unless You port Chromium to Your OS.

**Reply by tacitmarmot271** [reply]
OK I CAN WORK ON THAT its not easy but ill post a repo
